span.go

```diff
@@ -110,3 +112,18 @@ func InjectSpan(r *http.Request) *http.Request {
 	}
 	return r
 }
+
+// InjectHTTPTraceSpan Injects HttpTrace on the request
+func InjectHTTPTraceSpan(r *http.Request, trace *httptrace.ClientTrace) *http.Request {
```

Comment:
This seem super useful. So much that i don't see the point not doing it every time we call InjectSpan() .

Why not make this private and insert it in injectSpan

Reply:
also this `httptrace.ClientTrace` would nicely fit with the option pattern.
This would reduce signature to `InjectHTTPTraceSpan(r *http.Request, option Options...) ` and make it obvious that it is totally ok the not give any ClientTrace when calling this method.

Reply:
Combine these two and you won't have to add any line to the existing example to use this feature. 
Which mean still having super simple framework which is the point of this repo.

Reply:
Hello @FrancoisPoinsot,

I like the options pattern idea, but do you prefer the options pattern or the functional options pattern ?

Reply:
Now that i checked i only know what functional option pattern look like. No idea what option pattern is without these function call